### ISSUE TITLE
Problem retraining model with Softmax4D

### ISSUE THREAD

**Issue by dvro** [heuritech/convnets-keras#20]
I am trying to retrain the VGG16 for a specific set of classes.
It works with heatmap=False, (without Softmax4D), but I need to generate a heatmap.

Here is what I am doing:

- Removing the last 2 layers
- Freezing the remaining layers (trainable=False)
- Adding new layers (3 classes)
```
    model.add(Convolution2D(3,1,1,name="dense_3"))
    model.add(Softmax4D(axis=1,name="softmax"))
```
- Loading data and using preprocess_image_batch on images
- Finally, training the network
```
model.fit(X_, y_, nb_epoch=10, batch_size=4)
```
However, I have a problem finding the adequate format for y_ (the classes).

I know that y_ must have 3 dimensions `Exception: Error when checking model target: expected softmax to have 3 dimensions` but not sure how to do that considering that my y_ currently has 1 dimension (values 0,1 and 2).

Even applying `np_utils.to_categorical(y_, len(set(y_)))`, I still get only 2 dimensions.

Is the workflow correct? How can I solve this issue?

Thanks in advance,

**Comment by potholiday** [comment]
@dvro can you please tell me how do you retrained the  model for our data sets. This is my guess 

1. go to `/convnets-keras/convnetskeras/`

2. open `convnets.py`

3.  Edit `VGG_16 `function

4. Remove last layer

```
model.add(Dense(1000, name='dense_3'))
model.add(Activation("softmax",name="softmax"))
```

5.  Add a new layer (eg. for an yes/no output)

```
model.add(Dense(1, name='dense_3'))
model.add(Activation("softmax",name="softmax"))
```
6. Load  `vgg16_weights`

7. Retrain using `model.fit(X,Y)`

Am I in the right direction? Any help will be greatly appreciated.
